department-of-veterans-affairs/caseflow
# Aiman/CASEFLOW-4601-Create-EMO-Completed-Tab

Resolves [CASEFLOW-4601](https://vajira.max.gov/browse/CASEFLOW-4601)

### Description
- As a developer, to comply with the Education Pre-Docket requirements, I need to understand the expected behavior of the EMO queue and its tabs and develop them. The EMO tabs will be responsible for showing the appeals, worked by Education Pre-Docket, thru the pre-docket lifecycle.
- For EMO there will be three tabs: Unassigned, Assigned, and Completed
- There is no direct to user appeal assignment

### Acceptance Criteria
- Determination of the behavior and triggers of the Completed tab
- When an appeal is docketed by the BVA Intake organization, the appeal shows up on the EMO's completed tab.

### Testing Plan

1. Ensure that you have the edu_predocket_appeals feature toggle enabled. You can paste FeatureToggle.enable!(:edu_predocket_appeals) into the Rails console in order to enable it.
2. If you have not yet reseeded your database with the seed file updates from the feature branch, run make seed-dbs. The updated seed files are present in this branch.
3. Open Caseflow and switch to an Intake user.
4. Navigate to Caseflow Intake and begin the 10182 entry flow.
5. Whenever you arrive at the Add/Remove issues page, add an issue with the benefit type of 'Education' and select 'Yes' in the pre-docketing radio field.
6. Submit the appeal.
7. Switch to an EMO user.
8. Navigate to the EMO Organization's Queue. It should be [at this address](http://localhost:3000/organizations/edu-emo?tab=education_assigned&page=1).
9. Ensure that the appeal is not present. Whenever an appeal is first assigned to the EMO it will be present in the 'Unassigned' tab until an action is taken upon it.
10. Open up your Rails console.
11. c = Appeal.find_by_uuid("Insert Veterans appeal info from the URL in Caseflow here")
12.)
![image](https://user-images.githubusercontent.com/22192993/164768839-1038ffa1-a680-4791-8231-a4f0ad9c46b6.png)
12.) (Specify EMO task) emo_task_c = Task.find(7134)
13.) emo_task_c.update!(status: Constants.TASK_STATUSES.completed)
14.) 
![image](https://user-images.githubusercontent.com/22192993/164769105-3edc5580-e6a1-4c99-9832-3dae77574023.png)
15.) (Specify PreDocketTask) parent_task_c = Task.find(7133)
16.) parent_task_c.update!(status: Constants.TASK_STATUSES.completed)
17.) 
![image](https://user-images.githubusercontent.com/22192993/164769397-2a65f93c-5647-437a-b943-594809296351.png)
18.) At this point, the appeal in question will appear as completed in the EMO Completed tab

### Result of test file
**spec/models/queue_tabs/education_emo_completed_tasks_tab_spec.rb**
![image](https://user-images.githubusercontent.com/22192993/164768721-a7af43f2-cfe8-46c1-8a8c-d8d3240edc63.png)

###  User Facing Changes
<img width="1238" alt="Screen Shot 2022-04-27 at 3 40 22 PM" src="https://user-images.githubusercontent.com/18618189/165626253-dc74c673-9b51-43c6-9490-3bb40050cf09.png">
